src/inference/dev_api/openvino/runtime/icompiled_model.hpp

```diff
@@ -118,18 +119,21 @@ class OPENVINO_RUNTIME_API ICompiledModel : public std::enable_shared_from_this<
     std::shared_ptr<const ov::IPlugin> m_plugin;
     std::vector<ov::Output<const ov::Node>> m_inputs;
     std::vector<ov::Output<const ov::Node>> m_outputs;
+    bool m_is_from_cache = false;
 
     std::shared_ptr<ov::threading::ITaskExecutor> m_task_executor = nullptr;      //!< Holds a task executor
     std::shared_ptr<ov::threading::ITaskExecutor> m_callback_executor = nullptr;  //!< Holds a callback executor
 
     friend ov::CoreImpl;
+    friend ov::CompiledModel;
     friend ov::IExecutableNetworkWrapper;
     friend InferenceEngine::ICompiledModelWrapper;
 
     /**
      * @brief function allows to mark that model was loaded from cache
      */
     void loaded_from_cache();
+    bool is_loaded_from_cache();
```

Comment:
may be we need to remove this part from common ICompiledModel and all plugin should fill it by themselves?

Reply:
Yes removed from core